dashboard/src/containers/ResourceTableContainer/ResourceTableContainer.ts

```diff
@@ -0,0 +1,23 @@
+import { connect } from "react-redux";
+import { Action } from "redux";
+import { ThunkDispatch } from "redux-thunk";
+
+import actions from "../../actions";
+import ResourceTable from "../../components/AppView/ResourceTable/ResourceTable.v2";
+import ResourceRef from "../../shared/ResourceRef";
+import { IStoreState } from "../../shared/types";
+
+function mapStateToProps({ kube }: IStoreState) {
+  return {
+    resources: kube.items,
+  };
+}
+
+function mapDispatchToProps(dispatch: ThunkDispatch<IStoreState, null, Action>) {
+  return {
+    watchResource: (ref: ResourceRef) => dispatch(actions.kube.getAndWatchResource(ref)),
+    closeWatch: (ref: ResourceRef) => dispatch(actions.kube.closeWatchResource(ref)),
+  };
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(ResourceTable);
```

Comment:
> 1. Why are we creating a container here for the new resourceTable v2, when we could simply access the state in the function component as we've done elsewhere (using the `useState` hook)?

I guess you mean `useSelector`? Yes, we can use that to avoid the container. The methods are a bit more tricky though. In this case `watchResource` and `closeWatch` call actions and if we directly call those actions from within the component, we would not be able to easily mock them in the tests (even though we could). Let me give it a try.

> 2. This container isn't used afaict? I assume it's a later PR which connects it up - or am I missing where it was used?

Yep, it's not used in this PR, I have some temporary changes in my local env for `AppView` that uses the container. I will send the PR for that... eventually.

Reply:
[edit] I am finding some issues mocking those functions, will send a follow-up PR

Reply:
> > I'm pretty strongly against doing things like this, especially for a full two weeks.
> 
> Just to mention that I generally agree to avoid this (going back to write tests is a huge pain).

And it's not just going back to write tests - I need to keep working with the codebase while your gone requiring changes in components.

> I was willing to make an exception due to the tight schedule for a required demo and because I am going to be on vacations but anyway, let's do it the proper way.

I don't know the details, but if there's a required demo for you to record, then as mentioned, I'd work in a branch to make lots of progress without landing the code so you can record it without delay, then review and land the code properly in pieces when you're back.

If on the other hand, if you need to get the code landed for someone else to be able to do a demo, then yes we have a classic project management problem: time vs quality vs resources. I don't mind if you want to make an exception and sacrifice quality for time, but we need to be clear that since it's untested, it may break at any point (as I land other unrelated changes to `ResourceRef` in the mean-time or whatever). The other option is that I drop the multicluster work and switch over to help and continue while you're away, but I've not heard enough info to justify that (and people are waiting for that too). Let me know.